### PR TITLE
Add support to ignore keywords in go-mode

### PR DESCRIPTION
**Pull request by dakra**

### CODE REVIEW

File: symbol-overlay.el [hunk 1]
```diff
@@ -395,6 +396,14 @@ If SHOW-COLOR is non-nil, display the color used by current overlay."
                     (symbol-name keyword-symbol))))
     (string-match-p keyword symbol)))
 
+(defvar go-builtins)
+(defvar go-constants)
+(defvar go-mode-keywords)
+(defun symbol-overlay-ignore-function-go (symbol)
+  "Determine whether SYMBOL should be ignored (Go)."
+  (let ((keywords (append go-builtins go-constants go-mode-keywords)))
+    (string-match-p (concat "\\_<\\(" (mapconcat 'identity keywords "\\|") "\\)\\_>") symbol)))
```
[the remark this answers is not in the document]
Comment (dakra) [reply symbol-overlay.el]: Thanks, I'll change it.
It was just copy&paste programming from [here](https://github.com/dakra/symbol-overlay/blob/f34a2c6630fc426bbb3d0450ce58a5e7506221d3/symbol-overlay.el#L367).
I guess that can be changed too.

[the remark this answers is not in the document]
Comment (dakra) [reply symbol-overlay.el]: @purcell fixed. Please check again.

[the remark this answers is not in the document]
Comment (purcell) [reply symbol-overlay.el]: Ugh, right - didn't see those other bits. I'll maybe have a quick look and see if I can tidy up all those snippets.

[the remark this answers is not in the document]
Comment (purcell) [reply symbol-overlay.el]: Done in a88b79f - thanks1

**Pull request merged**